reputation-commitments/src/tests.rs

```diff
@@ -0,0 +1,399 @@
+		System::set_block_number(System::block_number() + 1);
+		System::on_initialize(System::block_number());
+	}
+}
+
+/// Progress blocks until the phase changes
+fn run_to_next_phase() {
+	let phase = EncointerScheduler::current_phase();
+	let mut blocknr = System::block_number();
+	while phase == EncointerScheduler::current_phase() {
+		blocknr += 1;
+		run_to_block(blocknr);
+	}
+}
+
+#[test]
+fn register_purpose_works() {
+	new_test_ext().execute_with(|| {
+		let alice = AccountId::from(AccountKeyring::Alice);
+		let some_description = DescriptorType::from_str("Some Description").unwrap();
+		let some_description2 = DescriptorType::from_str("Some Description 2").unwrap();
+		System::set_block_number(System::block_number() + 1); // this is needed to assert events
+		assert_ok!(EncointerReputationCommitments::do_register_purpose(some_description.clone()));
+		assert_eq!(EncointerReputationCommitments::current_purpose_id(), 1);
+		assert_eq!(EncointerReputationCommitments::purposes(0), some_description.clone());
+
+		// use extrinsic
+		assert_ok!(EncointerReputationCommitments::register_purpose(
+			RuntimeOrigin::signed(alice.clone()),
+			some_description2.clone()
+		));
+		assert_eq!(EncointerReputationCommitments::current_purpose_id(), 2);
+		assert_eq!(EncointerReputationCommitments::purposes(0), some_description);
+		assert_eq!(EncointerReputationCommitments::purposes(1), some_description2);
+
+		assert_eq!(
+			last_event::<TestRuntime>(),
+			Some(Event::RegisteredCommitmentPurpose(1, some_description2).into())
+		);
+	});
+}
+
+#[test]
+fn commitments_work() {
+	new_test_ext().execute_with(|| {
+		let mut ext = new_test_ext();
+		let alice = AccountId::from(AccountKeyring::Alice);
+		let bob = AccountId::from(AccountKeyring::Bob);
+		let cid = CommunityIdentifier::default();
+		let cid2 = CommunityIdentifier::new::<AccountId>(
+			Location { lat: Degree::from_num(0.1), lon: Degree::from_num(0.1) },
+			vec![],
+		)
+		.unwrap();
+		ext.insert(
+			participant_reputation((cid, 11), &alice),
+			Reputation::VerifiedUnlinked.encode(),
+		);
+		ext.insert(
+			participant_reputation((cid2, 12), &alice),
+			Reputation::VerifiedUnlinked.encode(),
+		);
+
+		ext.insert(participant_reputation((cid, 11), &bob), Reputation::VerifiedUnlinked.encode());
+
+		ext.execute_with(|| {
+			System::set_block_number(System::block_number() + 1); // this is needed to assert events
+
+			// register purposes
+			assert_ok!(EncointerReputationCommitments::do_register_purpose(
+				DescriptorType::from_str("Some Description").unwrap()
+			));
+			assert_ok!(EncointerReputationCommitments::do_register_purpose(
+				DescriptorType::from_str("Some Description2").unwrap()
+			));
+
+			// commit without hash
+			// alice
+			assert_ok!(EncointerReputationCommitments::do_commit_reputation(
+				&alice.clone(),
+				cid,
+				11,
+				0,
+				None
+			));
+			assert_eq!(EncointerReputationCommitments::commitments((cid, 11), (0, &alice)), None);
```

Comment:
this doesn't test anything: if value is None, it is the same as if the key were not there at all. you can change alice to Bob and it will still pass here

Reply:
absolutely, good catch.